### DISCUSSION TITLE
How to view messages?

### DISCUSSION THREAD

**Discussion by Wendel-de-Andrade** [YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor#69] [Q&A]
I managed to do the whole procedure and extract the Key and msgstore.db, I use Whatsapp Viewer to see the messages but it only accepts files with the "crypt" format, I renamed the file to see if it worked and the Key failed,
thinking that maybe renaming the file was giving this error I took the file "crypt" straight from the Wpp Backup folder and gave the same error. Is there a specific way to see the messages and I'm doing it wrong?

I don't know English, translated message.

Thank you all for the attention!

**Comment by YuvrajRaghuvanshiS** [comment]
You were right that renaming the file would give error. crypt extension means that it is encrypted file so you would need that file from WhatsApp folder in your phone. The reason it might have failed is because WhatsApp recently started rolling out "crypt14" instead of "crypt12" it has been using for years. I believe you have provided key path properly. For every encryption algorithm there are different steps to decrypt that, until developer of whatsapp viewer himself update his repo I guess we're stuck.

**Comment by GeorgKreuzmayr** [comment]
Hey,
awesome work you did here, thanks!
But are we still stuck with not being able to view the decrypted messages, as far as I can see WhatsappViewer is still not updated, right?
Also maybe does someone know if you could use the decrypted files to load them into WhatsApp on iOS?

**Reply by YuvrajRaghuvanshiS** [reply]
Hey,
Thank you for that.

WhatsappViewer should work as long we have plaintext database which this tool extracts. So I believe we're good here.
Talking of importing to iOS there is one other tool [watoi](https://github.com/residentsummer/watoi) which claims to do the same. I never tested it however (poor me, literally) so can't help if that does not work.
